src/commands/deploy/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { UserError } from "../../errors.js";
+import { CloudProviderIdentifier } from "../../models/cloudProviderIdentifier.js";
+import { CreateDatabaseRequest, GetDatabaseResponse } from "../../models/requests.js";
+import { DatabaseType } from "../../projectConfiguration/yaml/models.js";
+import {
+    createDatabase,
+    findLinkedDatabase,
+    getDatabaseByName,
+    linkDatabaseToEnvironment,
+} from "../../requests/database.js";
+import getProjectInfoByName from "../../requests/getProjectInfoByName.js";
+import { ProjectDetails, ProjectDetailsEnvElement } from "../../requests/models.js";
+import { createEmptyProject } from "../../requests/project.js";
+import { debugLogger } from "../../utils/logging.js";
+
+export async function getOrCreateEmptyProject(
+    projectName: string,
+    region: string,
+    stage: string = "prod",
+): Promise<{ project: ProjectDetails; projectEnv: ProjectDetailsEnvElement }> {
+    const project = await getProjectInfoByName(projectName);
+
+    if (!project) {
+        try {
+            await createEmptyProject({
+                projectName: projectName,
+                region: region,
+                cloudProvider: CloudProviderIdentifier.GENEZIO_CLOUD,
+                stage: stage,
+            });
+            debugLogger.debug(
+                `Project ${projectName} in region ${region} on stage ${stage} was created successfully`,
+            );
+        } catch (error) {
+            debugLogger.debug(`Error creating project ${projectName}: ${error}`);
+            throw new UserError(`Failed to create project ${projectName}.`);
+        }
```

Comment:
I find this cleaner, while also reducing indentation

```suggestion
        await createEmptyProject({
            projectName: projectName,
            region: region,
            cloudProvider: CloudProviderIdentifier.GENEZIO_CLOUD,
            stage: stage,
        }).catch(() => {
            debugLogger.debug(`Error creating project ${projectName}: ${error}`);
            throw new UserError(`Failed to create project ${projectName}.`);
        });
        debugLogger.debug(
            `Project ${projectName} in region ${region} on stage ${stage} was created successfully`,
        );
```